### ISSUE TITLE
Add a Remote::default_branch() function

### ISSUE THREAD

**Issue by gibfahn** [Byron/gitoxide#755]
### Summary 💡

Similar to the libgit2 [`Remote::default_branch()`](https://docs.rs/git2/latest/git2/struct.Remote.html#method.default_branch) function.

I believe under the hood this does something like:

```shell
git symbolic-ref --short refs/remotes/${remote_name}/HEAD
```

### Motivation 🔦

It would be nice to have an easy way to get the default branch for a remote (if known).

**Comment by Byron** [comment]
I see - it's actually trying to get the remotes `HEAD`, which typically lives as a tracking branch in the local repository as well. The `git2` implementation seems to query the remote to get the live view of the remote `HEAD` at least according to the method description.

It's worth noting that one can obtain these values today by probing the tracking branch for the remote `HEAD` for an offline version, or by asking for remote refs using the fetch machinery (it's no problem to just obtain remote refs based on a ref-spec without fetching or changing anything, see the implementation of `gix remote refs`.

**Comment by Byron** [comment]
Moved to [this Feature Proposal Discussion](https://github.com/GitoxideLabs/gitoxide/discussions/2792). This issue is being closed while the proposal is reviewed; it will become an issue again once review is complete.

_AI disclosure: This comment was written and posted by OpenAI Codex on behalf of Byron._
